client/src/assets/js/shared.js

```diff
@@ -48,4 +48,13 @@ function plot(plan, status_func) {
   }
 }
 
-export { readFile, readText, validate, plot, clearGraph };
+async function getPlan(id) {
+  try {
+    const response = await axios.post("/api/fetch/?id=" + id);
+    return response;
+  } catch (error) {
+    console.log(error);
```

Comment:
Might still need to return "something" here so it renders an error message on the screen?

Reply:
Raising alerts for now. Later might change it to Bootstrap modals or returning the error in the status window for better UX.